### PR TITLE
Move Kestrel shared source to src\Shared

### LINKED ISSUE #36035: Consider moving Kestrel shared source test attributes to src\Shared
e.g. https://github.com/dotnet/aspnetcore/blob/wtgodbe/3IIS/src/Servers/Kestrel/shared/test/TransportTestHelpers/HttpSysHttp3SupportedAttribute.cs and https://github.com/dotnet/aspnetcore/blob/wtgodbe/3IIS/src/Servers/Kestrel/shared/test/TransportTestHelpers/MsQuicSupportedAttribute.cs, which are used in tests across Kestrel, IIS, and HTTPSys. See https://github.com/dotnet/aspnetcore/pull/35934#discussion_r698708946

### PR DESCRIPTION
**Pull request by david-acker**
## Move Kestrel shared source to src\Shared

- Moves the following Kestrel shared source to src\Shared:
    - `HttpSysHttp3SupportedAttribute.cs`
    - `MsQuicSupportedAttribute.cs`
    - `TestResources.cs`
    - `\TestCertificates`

Fixes #36035

### CODE REVIEW

File: src/Servers/HttpSys/test/FunctionalTests/Microsoft.AspNetCore.Server.HttpSys.FunctionalTests.csproj [hunk 1]
```diff
@@ -22,8 +22,8 @@
     <Compile Include="$(SharedSourceRoot)InternalHeaderNames.cs" Linkbase="shared"/>
     <Compile Include="$(KestrelSharedSourceRoot)test\TestResources.cs" LinkBase="shared" />
```
Comment (Tratcher) [inline src/Servers/HttpSys/test/FunctionalTests/Microsoft.AspNetCore.Server.HttpSys.FunctionalTests.csproj:23]: What about these other kestrel shared sources that are being used outside of kestrel?

Comment (david-acker) [reply src/Servers/HttpSys/test/FunctionalTests/Microsoft.AspNetCore.Server.HttpSys.FunctionalTests.csproj]: I've moved remaining Kestrel shared sources (`TestResources.cs` and `\TestCertificates`) to `src\Shared`.

**Comment by Tratcher** [comment]
> CSC : error CS2001: Source file '/Users/runner/work/1/s/src/Servers/Kestrel/shared/test/TestResources.cs' could not be found. [/Users/runner/work/1/s/src/Servers/Kestrel/Transport.NamedPipes/test/Microsoft.AspNetCore.Server.Kestrel.Transport.NamedPipes.Tests.csproj]
##[error]CSC(0,0): error CS2001: (NETCORE_ENGINEERING_TELEMETRY=Build) Source file '/Users/runner/work/1/s/src/Servers/Kestrel/shared/test/TestResources.cs' could not be found.

https://dev.azure.com/dnceng-public/public/_build/results?buildId=141848&view=logs&j=48c285a2-daf4-5ecb-cb99-42232769ca29&t=1c30c545-4674-57a0-c50f-14816b5373e6&l=1125

**Comment by Tratcher** [comment]
src/Shared/TestCertificates/make-test-certs.sh 100755 → 100644? This might need to be chmond'd

**Comment by david-acker** [comment]
Interesting, good catch. I've added back execute permissions for that script.

**Review by Tratcher — APPROVED** [review]

**Comment by Tratcher** [comment]
Thanks

**Pull request merged**